.travis.yml

```diff
@@ -5,15 +5,24 @@ scala: 2.12.3
 
 jdk: oraclejdk8
 
-script: sbt ++$TRAVIS_SCALA_VERSION samples/compile test
+script:
+  - sbt ++$TRAVIS_SCALA_VERSION samples/compile test
+  - $(npm bin)/tsc samples/*.ts
 
 before_cache:
   # See http://www.scala-sbt.org/0.13/docs/Travis-CI-with-sbt.html
   # Tricks to avoid unnecessary cache updates
   - find $HOME/.sbt -name "*.lock" | xargs rm
   - find $HOME/.ivy2 -name "ivydata-*.properties" | xargs rm
 
+install:
+  - . $HOME/.nvm/nvm.sh
+  - nvm install stable
+  - nvm use stable
+  - npm install typescript
+
 cache:
   directories:
     - $HOME/.ivy2/cache
     - $HOME/.sbt/boot
+    - node_modules
```

Comment:
Is this standard practice? Is not dangerous? As in, if a new version of the CI script does not install some package anymore, it would still be in the cached `node_modules/`, and would therefore *still* be available, by accident?

Reply:
I think this is the standard practice.
[Caching Dependencies and Directories \- Travis CI](https://docs.travis-ci.com/user/caching/#Enabling-multiple-caching-features)
[The Travis CI Blog: Speed Up Your Builds: Cache Your Dependencies](https://blog.travis-ci.com/2013-12-05-speed-up-your-builds-cache-your-dependencies)
But I can understand your concerns as well. Do you want to delete it?

Reply:
OK let's keep it that way, then.